js/src/components/app-select-control/index.js

```diff
@@ -51,22 +53,22 @@ const AppSelectControl = ( props ) => {
 		...rest,
 	};
 
-	const hasSingleValueStyle = autoSelectFirstOption && options?.length === 1;
-	if ( hasSingleValueStyle ) {
+	const isNonInteractive =
+		( autoSelectFirstOption && options?.length === 1 ) || nonInteractive;
+	if ( isNonInteractive ) {
 		selectProps = {
 			...selectProps,
+			readOnly: true,
+			style: {
+				pointerEvents: 'none',
+			},
```

Comment:
Setting `pointer-events: none;` onto the `SelectControl` is different from to its wrapper. `SelectControl` component has the `label` prop allowing focus to the associated `<select>` element and then interacting it via keyboard.

https://github.com/user-attachments/assets/4af0962a-c6ed-429c-bdd8-f1cdf6d1e300

The original way also covers the `pointer-events: none;` to `<label>` element in order to avoid the issue. Although it won't be a problem in the scope of this PR, from the point of view of the component design and the fact that `AppSelectControl` is also a shared component, it should be reverted back to the original solution (but not including the concern in https://github.com/woocommerce/google-listings-and-ads/pull/2640#discussion_r1818466247).

Reply:
Sounds good. Updated @eason9487 https://github.com/woocommerce/google-listings-and-ads/pull/2640/commits/4cba145d913b80d949448ec3a32ee44afcc7d43a